### PR TITLE
Use SafeRoundtripper to fetch templates

### PR DESCRIPTION
**Pull request by vbrown608**
Updates template fetch to make use of https://github.com/netlify/netlify-commons/pull/82

### CODE REVIEW

File: mailme.go [hunk 1]
```diff
@@ -111,20 +112,20 @@ func (t *TemplateCache) Set(key, value string, expirationTime time.Duration) (*t
 }
 
 func (t *TemplateCache) fetchTemplate(url string, triesLeft int) (string, error) {
-	client := http.Client{
+	client := &http.Client{
 		Transport: &http.Transport{
```
Comment (mraerino) [inline mailme.go]: no need to set the transport. it's going to be replaced in the safehttp call anyways

File: mailme.go [hunk 2]
```diff
@@ -111,20 +112,20 @@ func (t *TemplateCache) Set(key, value string, expirationTime time.Duration) (*t
...
-			}).DialContext),
+			}).DialContext,
 			MaxIdleConns:          100,
 			IdleConnTimeout:       90 * time.Second,
 			TLSHandshakeTimeout:   10 * time.Second,
 			ExpectContinueTimeout: 1 * time.Second,
 		},
 	}
-
+	client = nfhttp.SafeHTTPClient(client, logrus.New())
```
Comment (mraerino) [inline mailme.go]: this should be getting a logger from a parent context

**Review by mraerino — COMMENTED** [review]
I think the best way to solve the log situation is to accept a logger in the constructor of the mailer and just pass it on

File: mailme.go (hunk 1 above)
Comment (vbrown608) [reply mailme.go]: The original transport gets wrapped by the new one. It's stored as `noLocalTransport.inner`.

However, Go's `DefaultTransport` is identical to the one defined here, except that the default has `ForceAttemptHTTP2: true`. Any idea if that's deliberate?

If it's not, we can just use the `DefaultTransport`. If it is, I think we should still start with the default and just update that one setting to make it explicit.

Comment (mraerino) [reply mailme.go]: agree on using `DefaultTransport` 👍

**Review by mraerino — COMMENTED** [review]
makes me wonder if we could just scrub the whole custom client creation and just pass the default http client into `SafeHTTPClient`?

**Comment by vbrown608** [comment]
Yep if we're going with the DefaultTransport I think we can definitely do that :)

**Comment by vbrown608** [comment]
Actually, we may want to specify a timeout on the default http client, like
```
http.Client{
  Timeout: time.Second * defaultTimeout,
}
```

Otherwise a malicious template URL could make us try to read the body forever.

**Comment by mraerino** [comment]
> Otherwise a malicious template URL could make us try to read the body forever, similar to the issue we had with accidentally-malicious humio in the analytics service.

this is actual a very limited attack vector because people could only provoke that for their own site and we don't care if people deliberately break it. i think we can ignore for now.

a timeout might even be weird for customers because they might have apis that take a long time to provide the template and since mailing is never in the hot path it would not influence latency of a service.

**Review by mraerino — APPROVED** [review]
Awesome 🚀

File: mailme.go [hunk 3]
```diff
@@ -111,20 +117,7 @@ func (t *TemplateCache) Set(key, value string, expirationTime time.Duration) (*t
...
-				DualStack: true,
-			}).DialContext),
-			MaxIdleConns:          100,
-			IdleConnTimeout:       90 * time.Second,
-			TLSHandshakeTimeout:   10 * time.Second,
-			ExpectContinueTimeout: 1 * time.Second,
-		},
-	}
-
+	client := nfhttp.SafeHTTPClient(http.DefaultClient, t.logger)
```
Comment (rybit) [inline mailme.go:120]: The problem with the default client is that it doesn't have any concepts of timeouts. I don't think it is a huge issue but something to be aware of.

Comment (vbrown608) [reply mailme.go:120]: See above comments :point_up:

**Pull request merged**